code/weapon/swarm.cpp

```diff
@@ -195,65 +155,34 @@ void swarm_update_direction(object *objp)
 
 	wp = &Weapons[objp->instance];
 
-	if (wp->swarm_index == -1) {
-		return;
-	}
-
 	wip = &Weapon_info[wp->weapon_info_index];
 	hobjp = wp->homing_object;
 	pi = &Objects[wp->objnum].phys_info;
-	swarmp = &Swarm_missiles[wp->swarm_index];
 
-	// check if homing is lost.. if it is then get a new path to move swarm missile along
+	// check if homing is lost.. if it is then set the target point dead ahead
 	if ( swarmp->homing_objnum != -1 && hobjp == &obj_used_list ) {
 		swarmp->change_timestamp = 1;
-		swarmp->path_num = -1;
 		swarmp->homing_objnum = -1;
+		swarm_set_defaut_target_pos(objp, swarmp);
```

Comment:
Just double checking, did you check if the behavior matches if the target is lost, too?

Reply:
Yes, even as largely irrelevant I might think it to be, this was done specifically the maintain behavior on lost target.